plugin.rb

```diff
@@ -151,4 +151,17 @@ def create_json(obj)
     mount ::DiscourseStaffNotes::Engine, at: "/staff_notes"
   end
 
+  Warning.class_eval do
+    after_create :create_staff_note
+
+    protected
+    def create_staff_note
+      user = User.find_by_id(self.user_id)
+      created_by_user = User.find_by_id(self.created_by_id)
+      warning_topic = Topic.find_by_id(self.topic_id)
+      raw_note = I18n.t("staff_notes.official_warning", username: created_by_user.username, warning_link: "[#{warning_topic.title}](#{warning_topic.url})")
```

Comment:
ahh nvm XD I'll look like `User.where(id: self.created_by_id).pluck(:username).first` which is worst